packages/editor/src/components/post-publish-button/index.js

```diff
@@ -118,7 +118,7 @@ export class PostPublishButton extends Component {
 			( ! isPublishable && ! forceIsDirty );
 
 		let publishStatus;
-		if ( ! hasPublishAction ) {
+		if ( ! isPublished && ! isBeingScheduled && ! hasPublishAction ) {
```

Comment:
I think this also needs to check for `! isCurrentPostScheduled` like https://github.com/WordPress/gutenberg/blob/4b9fc094123e5dd426bb12bf5f64e729714c7544/packages/edit-post/src/components/header/post-publish-button-or-toggle.js#L55, right @youknowriad?

Reply:
Thanks for `isCurrentPostScheduled` hint!

I'm still investigating this but it seems like for a regular author `isCurrentPostScheduled()` and `isEditedPostBeingScheduled()` both always return `true` once a scheduled post is saved.
For a user with only `edit_posts` and `edit_published_posts` caps both are returning false...

Reply:
They just check the saved and edited `status` property of the post respectively.